scripts/solve.js

```diff
@@ -152,3 +152,19 @@ function runStaticAnalysis(code) {
       The total cyclomatic compexity is ${totalComplexity}.`
   document.getElementById('analysis-output').innerText = metricsString;
 }
+
+function submitSolution() {
+  const submitButton = document.getElementById('submit-button');
+  submitButton.disabled = true;
+  submitButton.innerHTML = 
+    '<span class="spinner-border spinner-border-sm" role="status" aria-hidden="true"></span>Loading...';
+  const solution = new Blob([JSON.stringify({code: codeMirror.getValue()})], {type: 'application/json'});
+
+  fetch(`${location.pathname}`, {method: 'POST', body: solution}).then((res) => {
+    return res.json();
+  }).then((errors) => {
```

Comment:
I might be mistaken but i thought the first function parameter in a `.then` is for the success case.

Also, what would happen here if the post request earlier failed? Is re-enabling the button something that should always happen? If so, is it better represented as a `.finally`?

Reply:
sorry, i should have better specified here as well, but in the TODO i documented that when this branch is integrated the server will evaluate the code. if the evaluated code is incorrect, it should return the errors to the front end in a JSON (which is what `errors` represents, otherwise it will redirect to the rankings page.

Reply:
alright, your answer to the first point makes sense

for the second point, imagine this scenario. User clicks button to submit, button is disabled. While running the POST request to the backend, it times out. The button is never re-enabled so the user cannot easily retry. That's why I thought it might be better to use `.finally`.

Reply:
that makes sense, thank you. i've pushed the changes now!